docs/user/reconciliation.md

```diff
@@ -164,3 +164,10 @@ For subsequent updates, the builder blocks until:
 ### Job
 
 After pushing the merged Job into the cluster, the builder blocks until the Job succeeds.
+
+Since neither the use case requiring a Job's spec.selector to be explicitly set nor the handling thereof is well defined the code will panic if a user attempts to set spec.selector.
```

Comment:
Maybe rephrase to:

> The cluster-version operator will panic if `spec.selector` is set, because `selector` is immutable and there are no clear use-cases for setting it in release manifests.

or some such.  That leads with the behavior, so folks who don't care about the reasoning can skip over it more easily.

Reply:
Rephrased for clarity but left out reference to immutable which really has nothing to do with a user explicitly setting the spec.selector in the manifest. The fact it is immutable is relevant if CVO were to attempt an update so that's why I included it under 'Subsequent updates'.

Reply:
6b58e7a wording looks good to me.